### PR TITLE
Imagen

### PR DESCRIPTION
**Pull request by AGrigis**
To process the IMAGEN data with 'pypreprocess' I had to make the following modifications:

**pypreprocess/configure_spm**: 
- change the logger name: __main__ to 'pypreprocess' (rename this logger because it will be use during the processings).
- add parameters to the log configuration function in order to be able to redirect properly the log in a file.
- fix the SPM standalone command: 'run script' is not working with the last i2bm SPM12 standalone version and 'run' is deprecated.

**pypreprocess/subject_data**:
- add a parameter to say if we want to setup or not the caching mechanism, and create the cache directory accordingly.

**pypreprocess/nipype_preproc_spm_utils**:
- add a parameter to force the EPI/T1 template.
- log the interface failures.
- allow to setup custom or to overwrtie existing nipype interface parameters.
- if no coregistration is performed, change the normalization source to the mean fMRI volume.
- fix minor issues when the smart caching is disabled.

### CODE REVIEW

**Comment by bthirion** [comment]
Sorry, the PR is long and I'm travelling. I'll be reviewing when I can. 
Meanwhile, can you take a look at the travis failure ?

**Comment by AGrigis** [comment]
The Travis failure is related to a deprecated colormap not related to this PR.
I thing PR #304 deals with this issue.

**Comment by KamalakerDadi** [comment]
>Meanwhile, can you take a look at the travis failure ?

>The Travis failure is related to a deprecated colormap not related to this PR.
>I thing PR #304 deals with this issue.

Upgrading the Nilearn in the travis enviroment should fix the travis failure. I can open a new PR by upgrading Nilearn and see if we have fixed this.

File: pypreprocess/nipype_preproc_spm_utils.py [hunk 1]
```diff
@@ -114,10 +134,10 @@ def _configure_backends(spm_dir=None, matlab_exec=None, spm_mcr=None,
 
 
 def _do_subject_slice_timing(subject_data, TR, TA=None, spm_dir=None,
-                             matlab_exec=None, spm_mcr=None, ref_slice=0,
+                             matlab_exec=None, spm_mcr=None, ref_slice=1,
```
Comment (bthirion) [inline pypreprocess/nipype_preproc_spm_utils.py]: Can you explain why we need ref_slice=1 ?

File: pypreprocess/nipype_preproc_spm_utils.py [hunk 2]
```diff
@@ -148,6 +168,7 @@ def _do_subject_slice_timing(subject_data, TR, TA=None, spm_dir=None,
     assert 1 <= ref_slice <= nslices, ref_slice
 
     # compute slice indices / order
+    ref_slice -= 1
```
Comment (bthirion) [inline pypreprocess/nipype_preproc_spm_utils.py]: Oh, here is the answer I assume ?

**Comment by bthirion** [comment]
@AGrigis  OK for merging in spite of CircleCI failure (unrelated AFAIK)

File: pypreprocess/nipype_preproc_spm_utils.py (hunk 1 above)
Comment (AGrigis) [reply pypreprocess/nipype_preproc_spm_utils.py]: I decided to set this parameter to 1 in order to have the same default values in all functions: check for instance the do_subject_preproc function defaults.
On to of that the test on line 148 (master) / 168 (PR) is not valid if initialized with the default ref_slice value: assert 1 <= ref_slice <= nslices, ref_slice

File: pypreprocess/nipype_preproc_spm_utils.py (hunk 2 above)
Comment (AGrigis) [reply pypreprocess/nipype_preproc_spm_utils.py]: I converted here the ref_slice form Matlab to Python index.
Indeed the following of the code expects a Python ref_slice starting at zero.

**Pull request merged**